axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineCreateServiceImpl.java

```diff
@@ -493,133 +501,149 @@ public List<MoveLine> createMoveLines(
-                partnerAccountingSituation.getFullName(),
-                company.getCode());
-          }
-          int vatSystemSelect =
-              accountingSituationService.determineVatSystemSelect(
-                  accountingSituation, invoiceLineTax);
-          account =
-              taxAccountService.getAccount(
-                  tax,
-                  company,
-                  move.getJournal(),
-                  vatSystemSelect,
-                  false,
-                  move.getFunctionalOriginSelect());
-          if (account == null) {
-            throw new AxelorException(
-                move,
-                TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
-                I18n.get(AccountExceptionMessage.MOVE_LINE_6),
-                tax.getName(),
-                company.getName());
-          }
-
           moveLine =
-              this.createMoveLine(
+              this.createMoveLineFromInvoiceLineTax(
+                  invoiceLineTax,
+                  invoice,
+                  accountingSituation,
                   move,
+                  company,
                   partner,
-                  account,
-                  invoiceLineTax.getSubTotalExcludingFixedAssets(),
-                  invoiceLineTax.getCompanySubTotalExcludingFixedAssets(),
-                  null,
-                  !isDebitCustomer,
-                  invoice.getInvoiceDate(),
-                  null,
-                  invoice.getOriginDate(),
-                  moveLineId++,
+                  tax,
                   origin,
-                  null);
-          moveLine.setTaxLineSet(Sets.newHashSet(invoiceLineTax.getTaxLine()));
-          moveLine.setTaxRate(invoiceLineTax.getTaxLine().getValue());
-          moveLine.setTaxCode(tax.getCode());
-          moveLine.setVatSystemSelect(invoiceLineTax.getVatSystemSelect());
+                  moveLineId,
+                  isDebitCustomer,
+                  false);
+
           moveLines.add(moveLine);
+          moveLineId++;
         }
       }
     }
 
+    this.consolidateTaxMoveLines(moveLines);
+
     if (consolidate) {
       moveLineConsolidateService.consolidateMoveLines(moveLines);
     }
 
     return moveLines;
   }
 
+  protected MoveLine createMoveLineFromInvoiceLineTax(
+      InvoiceLineTax invoiceLineTax,
+      Invoice invoice,
+      AccountingSituation accountingSituation,
+      Move move,
+      Company company,
+      Partner partner,
+      Tax tax,
+      String origin,
+      int moveLineId,
+      boolean isDebitCustomer,
+      boolean isFixedAsset)
+      throws AxelorException {
+    int vatSystemSelect =
+        accountingSituationService.determineVatSystemSelect(accountingSituation, invoiceLineTax);
+
+    Account account =
+        taxAccountService.getAccount(
+            tax,
+            company,
+            move.getJournal(),
+            invoiceLineTax.getInvoiceLine().getAccount(),
```

Comment:
There will be an NPE if I ventilate an old Invoice that has been created before the 8.1